src/lfo.rs

```diff
@@ -0,0 +1,69 @@
+use crate::ring_buffer::RingBuffer;
+
+pub struct WaveTableLfo {
+    wave_table: RingBuffer<f32>, //  using RingBuffer to store the wavetable
+    mod_freq: usize,
+    amplitude: f32,
+    sample_freq: usize,
+
+}
+
+impl WaveTableLfo {
+    pub fn new(sample_freq: usize, mod_freq: usize, amplitude: f32) -> Self {
+
+        let mut wave_table_size = sample_freq;
+        let mut wave_table = RingBuffer::new(wave_table_size);
+
+        // Populate the wavetable with a single cycle of a sine wave
+        for i in 0..wave_table_size {
+            let phase =  (i / wave_table_size)  as f32  * 2.0 * std::f32::consts::PI;
+            wave_table.push(phase.sin());
+        }
+
+        WaveTableLfo {
+            wave_table,
+            mod_freq,
+            amplitude,
+            sample_freq,
+
+        }
+    }
+    pub fn get_value_at_index(&self, index: usize) -> f32 {
+        self.amplitude * self.wave_table.get(index)
+    }
+
+    pub fn set_value(&mut self, param: &str, value: f32) {
+        match param {
+            "mod_freq" =>{
+                self.mod_freq = value as usize;
+            },
+            "amplitude" => {
+                self.amplitude = value;
+            },
+            _ => unimplemented!(),
+        }
+    }
+
+
+
+}
+
+
+#[cfg(test)]
+mod tests{
+    use super::*;
+
+    #[test]
+    fn test_implementation(){
```

Comment:
It doesn't seem like a correct test function here. You might need to have a sentence with "assert" to make the output the same as the one you assume I think.